data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v2/validator/PostAggregateAndProofsV2IntegrationTest.java

```diff
@@ -0,0 +1,156 @@
+            spec.getGenesisSchemaDefinitions()
+                .getSignedAggregateAndProofSchema()
+                .getJsonTypeDefinition());
+  }
+
+  @TestTemplate
+  void shouldPostAggregateAndProofs_NoErrors() throws Exception {
+    when(validatorApiChannel.sendAggregateAndProofs(anyList()))
+        .thenReturn(SafeFuture.completedFuture(Collections.emptyList()));
+
+    final List<SignedAggregateAndProof> aggregateAndProofs =
+        List.of(
+            dataStructureUtil.randomSignedAggregateAndProof(),
+            dataStructureUtil.randomSignedAggregateAndProof());
+
+    final Response response =
+        post(
+            PostAggregateAndProofsV2.ROUTE,
+            JsonUtil.serialize(aggregateAndProofs, aggregateAndProofsListTypeDef),
+            Collections.emptyMap(),
+            Optional.of(specMilestone.name().toLowerCase(Locale.ROOT)));
+
+    assertThat(response.code()).isEqualTo(SC_OK);
+    assertThat(response.body().string()).isEmpty();
+  }
+
+  @TestTemplate
+  void shouldReturnErros() throws Exception {
+    final SubmitDataError firstSubmitDataError =
+        new SubmitDataError(UInt64.ZERO, "Bad aggregate and proofs");
+    when(validatorApiChannel.sendAggregateAndProofs(anyList()))
+        .thenReturn(SafeFuture.completedFuture(List.of(firstSubmitDataError)));
+    final List<SignedAggregateAndProof> aggregateAndProofs =
+        List.of(
+            dataStructureUtil.randomSignedAggregateAndProof(),
+            dataStructureUtil.randomSignedAggregateAndProof());
+    final Response response =
+        post(
+            PostAggregateAndProofsV2.ROUTE,
+            JsonUtil.serialize(aggregateAndProofs, aggregateAndProofsListTypeDef),
+            Collections.emptyMap(),
+            Optional.of(specMilestone.name().toLowerCase(Locale.ROOT)));
+    assertThat(response.code()).isEqualTo(SC_BAD_REQUEST);
+
+    final JsonNode resultAsJsonNode = JsonTestUtil.parseAsJsonNode(response.body().string());
+    assertThat(resultAsJsonNode.get("message").asText()).isEqualTo(PARTIAL_PUBLISH_FAILURE_MESSAGE);
+  }
+
+  @TestTemplate
+  void shouldFailWhenMissingConsensusHeader() throws Exception {
+    when(validatorApiChannel.sendAggregateAndProofs(anyList()))
+        .thenReturn(SafeFuture.completedFuture(Collections.emptyList()));
+
+    final List<SignedAggregateAndProof> aggregateAndProofs =
+        List.of(
+            dataStructureUtil.randomSignedAggregateAndProof(),
+            dataStructureUtil.randomSignedAggregateAndProof());
+
+    final Response response =
+        post(
+            PostAggregateAndProofsV2.ROUTE,
+            JsonUtil.serialize(aggregateAndProofs, aggregateAndProofsListTypeDef));
+
+    assertThat(response.code()).isEqualTo(SC_BAD_REQUEST);
+
+    final JsonNode resultAsJsonNode = JsonTestUtil.parseAsJsonNode(response.body().string());
+    assertThat(resultAsJsonNode.get("message").asText())
+        .isEqualTo("(Eth-Consensus-Version) header value was unexpected");
```

Comment:
if this header is missing and we require it, i'd expect a more clear message, this is very ambiguous... i'd expect a message along the lines of 'missing required header value for Eth-Consensus-Version'...

Reply:
Made it clearer when missing: `Missing required header value for (Eth-Consensus-Version)`